### ISSUE TITLE
Where do I get `bpe_models/wikitext-2.bpe.32000`?

### ISSUE THREAD

**Issue by darkmatter08** [lightonai/dfa-scales-to-modern-deep-learning#2]
I'm trying to run `train_lm.py` to replicate Table 5 in your paper. For the simple baseline experiment `python train_lm.py --gpu_id 0 --beta2 0.98` I am missing the file `bpe_models/wikitext-2.bpe.32000`. How do I download or generate this file? Also the default `--dataset` is 'wikitext103', not 'wikitext2', is this compatible with `bpe_models/wikitext-2.bpe.32000`? 

Thanks! And cool project!

**Comment by slippylolo** [comment]
Hey @darkmatter08, 

Thank you for your interest in our paper!

The `.bpe` file refer to the tokenizer used for generating dataset-dependent tokens, as first described in https://arxiv.org/abs/1508.07909.

You will need to build the tokenizer yourself against the dataset you use. To do so, you can run the following snippet: 
```
import youtokentome as yttm
vocab_size = 32000
train_data_path = ".data/wikitext-103/wikitext-103/wiki.train.tokens"  # Path to the torchtext dataset
model_path = f"bpe_models/wikitext-103.bpe.{vocab_size}"  # Where to save the tokenizer

yttm.BPE.train(data=train_data_path, vocab_size=vocab_size, model=model_path)
```

The `.bpe` file necessary to run our Transformer will be saved automatically at `model_path`. 

There is indeed a mismatch in the default configuration: you should use a tokenizer (`.bpe`) that matches the dataset you wish to use. (Mismatching the tokenisers might work, but will probably lead to subpar performance.) 

Thank you for noticing, we will fix that :).

**Comment by darkmatter08** [comment]
Thanks! For other readers, I downloaded the files from [here](https://blog.einstein.ai/the-wikitext-long-term-dependency-language-modeling-dataset/) and unzipped them into `.data/wikitext-103/`, in the `dfa-scales-to-modern-deep-learning/paper-experiments/natural-language-processing/` directory. I then executed @slippylolo 's script, and then was able to run `python train_lm.py --gpu_id 0 --beta2 0.98`. I also had to create an empty folder the script expected to exist. 
Separately, I also had to massage the requirements.txt file to get it to install on my system, and I had to use python3.6 to get things running (python3.5 didn't work).
